File: src/Goods.BackOffice/client-app/src/domain/products/product.ts
import { Page } from '../../tools/types/page';
import { ProductCategory } from './productCategory';

export class Product {
	constructor(
		public readonly id: string,
		public readonly category: ProductCategory,
		public readonly name: string,
		public readonly description: string,
		public readonly price: number
	) { }
}

export function mapToProductsPage(data: any): Page<Product> {
	return Page.convert(data, mapToProduct);
}

export function mapToProducts(data: any[]): Product[] {
	return data.map(mapToProduct);
}

export function mapToProduct(data: any): Product {
	return new Product(data.id, data.category, data.name, data.description, data.price);
}

File: src/Goods.Domain/Products/Product.cs
namespace Goods.Domain.Products;

public class Product
{
    public Guid Id { get; }
    public ProductCategory Category { get; }
    public String Name { get; }
    public String? Description { get; }
    public Decimal Price { get; }
    public Boolean IsRemoved { get; }

    public Product(Guid id, ProductCategory category, String name, String? description, Decimal price, Boolean isRemoved)
    {
        Id = id;
        Category = category;
        Name = name;
        Description = description;
        Price = price;
        IsRemoved = isRemoved;
    }
}